### ISSUE TITLE
PomTask fails with Gradle 3.4 when used with the java-library plugin

### ISSUE THREAD

**Issue by ddimtirov** [stempler/gradle-versioneye-plugin#34]
Gradle 3.4 introduced the notion of unresolvable configs:

https://docs.gradle.org/3.4/release-notes.html#configurations-can-be-unresolvable 

In particular, when used with the new `java-library` plugin, the versioneye:1.6 plugin fails the build witht he following exception:

https://scans.gradle.com/s/ytj7atrn5zzr4/failure 

```
Caused by: java.lang.IllegalStateException: Resolving configuration 'api' directly is not allowedClose stacktrace
at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.assertResolvingAllowed(DefaultConfiguration.java:819)
at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.resolveToStateOrLater(DefaultConfiguration.java:419)
at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.getResolvedConfiguration(DefaultConfiguration.java:414)
at org.standardout.gradle.plugin.versioneye.PomTask$_create_closure1_closure6.doCall(PomTask.groovy:79)
at org.standardout.gradle.plugin.versioneye.PomTask$_create_closure1.doCall(PomTask.groovy:76)
at org.standardout.gradle.plugin.versioneye.PomTask.create(PomTask.groovy:75)
```

Even as a short-term solution, if the versioneye plugin supported the [`java-library` configurations](https://docs.gradle.org/3.4/userguide/java_library_plugin.html#sec:java_library_configurations_graph), that would be useful.

**Comment by stempler** [comment]
So the easy way to fix this seems to be just ignoring unresolvable configurations. Or what reasons would you see not to? The dependencies defined there should be available in another resolvable configuration as well, or not?

What do you mean exactly with supporting the java-library configurations? In what sense should they be treated special?

**Comment by ddimtirov** [comment]
From what I understand, ignoring the unresolvable configurations should be sufficient. 

If that would be implemented, you can ignore the java-library part of my request (it was there as a priority highlight, in case we start a lenghty discussion on "the right thing to do").

**Comment by johnjohndoe** [comment]
It looks like I am facing the same issue in two projects of mine [HalfnarpClient (Java library)](https://github.com/johnjohndoe/HalfnarpClient), [c3media-base (Kotlin library)](https://github.com/johnjohndoe/c3media-base):

> Resolving configuration 'apiElements' directly is not allowed

**Comment by ddimtirov** [comment]
@johnjohndoe until a new version of the plugin comes out you can use the following workaround:

```
versioneye {
    ... your settings here
    exclude project.configurations.findAll { !it.canBeResolved }*.name as String[]
}
```

**Comment by johnjohndoe** [comment]
@ddimtirov Thank you for posting the workaround. It works for me. The other option is to temporarily downgrade the Gradle wrapper.

**Comment by mmoayyed** [comment]
@ddimtirov Thanks very much. Ran into the same issue while evaluating [CAS](https://github.com/apereo/cas).

**Comment by johnjohndoe** [comment]
@ddimtirov Your workaround works for me with Gradle 3.5 as well.

**Comment by stempler** [comment]
Can you try again with the [current snapshot](https://github.com/stempler/gradle-versioneye-plugin#using-the-current-snapshot)?
The reason why it fails even if the workaround is used is most likely that subprojects are included and there are configurations that are not resolvable that don't occur in the main project.

**Comment by sbuettner** [comment]
@stempler Just did a successful test with the snapshot version in a kotlin project using `Gradle 4.1-20170707032407+0000` that had the same issue before.

**Comment by johnjohndoe** [comment]
@stempler Confirmed from my side. Tested with a project which used `gradle-versioneye-plugin:1.4.0` up to now.

**Comment by stempler** [comment]
Fix contained in 1.5.0 release.
